ledger-core/v2/virtual/object/object.go

```diff
@@ -321,6 +338,32 @@ func (sm *SMObject) stepWaitIndefinitely(ctx smachine.ExecutionContext) smachine
 	return ctx.Sleep().ThenRepeat()
 }
 
+func (sm *SMObject) createWaitOrderedPendingTableSM(ctx smachine.ExecutionContext) {
+	syncSM := SMAwaitTableFill{
+		sync: sm.OrderedPendingListFilled,
+	}
+
+	// syncSM acquire OrderedExecute semaphore in init step.
+	ctx.InitChildWithPostInit(func(ctx smachine.ConstructionContext) smachine.StateMachine {
+		return &syncSM
+	}, func() {
+		sm.OrderedPendingListFilledCallback = syncSM.stop
+	})
+}
+
+func (sm *SMObject) createWaitUnorderedPendingTableSM(ctx smachine.ExecutionContext) {
+	syncSM := SMAwaitTableFill{
+		sync: sm.UnorderedPendingListFilled,
+	}
+
+	// syncSM acquire OrderedExecute semaphore in init step.
```

Comment:
wrong comment. Those ordered/unordered through switch/case cause lot off problems and copy-paste code =(

Reply:
truw, we'll need to figure out how to proper structure Info with this semaphores, counters and earliest pulses

Reply:
fixed comment